packages/expect/src/matchers.ts

```diff
@@ -277,7 +277,11 @@ const matchers: MatchersObject = {
     return {message, pass};
   },
 
-  toBeInstanceOf(this: MatcherState, received: any, expected: Function) {
+  toBeInstanceOf(
+    this: MatcherState,
+    received: new () => unknown,
```

Comment:
I don't think these are safe, I'd prefer to use `unknown` and narrow down. We're supposed to throw if they are called with the wrong type anyways, and making those checks narrow the type seems ideal. These get called by user code who provides whatever they feel like.

Reply:
Yeah you're totally right!